src/ucp/dt/dt_contig.c

```diff
@@ -21,3 +21,39 @@ size_t ucp_memcpy_pack(void *dest, void *arg)
     UCS_PROFILE_CALL(memcpy, dest, ctx->src, length);
     return length;
 }
+
+
+static UCS_F_ALWAYS_INLINE void
+ucp_dt_contig_pack_internal(ucp_worker_h worker, void *dest, const void *src,
+                            size_t length, ucs_memory_type_t mem_type,
+                            ucp_dt_pack_func_t pack)
+{
+    pack(worker, dest, src, length, mem_type);
+}
+
+
+void ucp_dt_contig_pack(ucp_worker_h worker, void *dest, const void *src,
+                        size_t length, ucs_memory_type_t mem_type)
+{
+    ucp_dt_pack_func_t pack = ucp_dt_get_pack_func(mem_type);
+
+    ucp_dt_contig_pack_internal(worker, dest, src, length, mem_type, pack);
```

Comment:
not sure the compiler will inline it this way, can you check it?
should be something like
```
if (mem_type == HOST) {
    ucp_dt_contig_pack_internal(worker, dest, src, length, mem_type, pack_host);
} else {
    ucp_dt_contig_pack_internal(worker, dest, src, length, mem_type, pack_memtype);
}
```